client/dashboard/src/components/sources/SourceCard.tsx

```diff
@@ -35,8 +42,19 @@ export function SourceCard({
   handleViewAsset: (assetId: string) => void;
   setChangeDocumentTargetSlug: (slug: string) => void;
 }) {
+  const routes = useRoutes();
   const IconComponent = asset.type === "openapi" ? FileCode : SquareFunction;
 
+  const handleCardClick = () => {
```

Comment:
We should wrap the card in a `Link` component rather than navigating in an on-click handler, so we get proper link behavior (eg. right click opens in new tab)